torch/_functorch/partitioners.py

```diff
@@ -1680,18 +1690,28 @@ def get_recomputable_banned_nodes(banned_nodes: List[fx.Node]) -> List[fx.Node]:
     ]
     from torch.utils._mode_utils import no_dispatch
 
-    def get_saved_values_knapsack(memory_budget):
+    def get_saved_values_knapsack(memory_budget, node_info, joint_graph):
         with no_dispatch():
             (
                 expected_runtime,
                 saved_node_idxs,
                 recomputable_node_idxs,
             ) = _optimize_runtime_with_given_memory(
-                memories_banned_nodes, runtimes_banned_nodes, max(memory_budget, 0)
+                joint_graph,
+                memories_banned_nodes,
+                runtimes_banned_nodes,
+                max(memory_budget, 0),
+                node_info,
+                all_recomputable_banned_nodes,
             )
         dont_ban = set()
         for idx in recomputable_node_idxs:
-            dont_ban.add(all_recomputable_banned_nodes[idx])
+            # if idx in all_recomputable_banned_nodes:
+            try:
+                dont_ban.add(all_recomputable_banned_nodes[idx])
+            except BaseException:
```

Comment:
What is the expected exception here? Are we expected an index error?

I am a bit surprised you can't use:

```
            # if idx in all_recomputable_banned_nodes:
```

Reply:
@basilwong - we can use that, but I think it adds an O(n) operation inside the larger for loop. This operation actually broke one of the pre-existing unit tests, because the flop count increased ....

Reply:
(The expected exception here is that idx is not in all_recomputable_banned_nodes)